### PR TITLE
DOC: Fix method documentation of function sort in MaskedArray

### LINKED ISSUE #15841: DOC: numpy.ma.sort and copying behaviour
I started out thinking I've found a docbug but I'm becoming less sure. The problem is that [the documentation of `numpy.ma.sort`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ma.sort.html) ([devdocs version](https://numpy.org/devdocs/reference/generated/numpy.ma.sort.html#numpy.ma.sort)) says
> Sort the array, in-place
> [...]
> Returns: `sorted_array`

Whereas the behaviour seems to be the opposite:
```py
>>> import numpy as np
>>> arr = np.arange(5)[::-1]
>>> np.ma.sort(arr)
array([0, 1, 2, 3, 4])
>>> arr
array([4, 3, 2, 1, 0])
```

The weird thing is that I've found https://github.com/numpy/numpy/issues/1966 which suggests that until 2010 the docs had mentioned a copy and the behaviour was mutating, and the docstring was subsequently fixed. However looking at the git blame even [in commit 5cb370e923 from 2008](https://github.com/numpy/numpy/blob/5cb370e9234582f62231384c26a426301dad3331/numpy/ma/core.py#L2859) I can see
```py
from numpy import array as narray
# [...]
def sort(...):
    a = narray(a, copy=True, subok=True)
```
This seems to me that even in 2008 it should have returned a copy, which makes the linked docbug issue very confusing. What am I missing?

Ultimately I think this is a docbug, because `np.ma.sort` should match `np.sort` which copies (and is documented as such), whereas the corresponding instance methods should sort in-place.

### PR DESCRIPTION
**Pull request by vrakesh**
closes #15841

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -6824,7 +6824,13 @@ def argsort(a, axis=np._NoValue, kind=None, order=None, endwith=True, fill_value
 argsort.__doc__ = MaskedArray.argsort.__doc__
 
 def sort(a, axis=-1, kind=None, order=None, endwith=True, fill_value=None):
-    "Function version of the eponymous method."
+    """
+    Equivalent to self.copy.sort()
```
Comment (seberg) [inline numpy/ma/core.py:6828]: Maybe start with `Return a sorted copy of the masked array.` And then continue with the equivalent to `...` and ```refer to `numpy.ma.MaskedArray.sort` ``` for the full documentation.
Also `self.copy().sort()`, although I am not sure it actually helps much to spell it out.

Comment (vrakesh) [reply numpy/ma/core.py]: Agreed, made the change.

File: numpy/ma/core.py [hunk 2]
```diff
@@ -6824,7 +6824,16 @@ def argsort(a, axis=np._NoValue, kind=None, order=None, endwith=True, fill_value
 argsort.__doc__ = MaskedArray.argsort.__doc__
 
 def sort(a, axis=-1, kind=None, order=None, endwith=True, fill_value=None):
-    "Function version of the eponymous method."
+    """
+    Return a sorted copy of the masked array.
+
+    Equivalent to creating an copy of the array
```
Comment (anirudh2290) [inline numpy/ma/core.py:6830]: nit: 
```suggestion
    Equivalent to creating a copy of the array
```

File: numpy/ma/core.py [hunk 3]
```diff
@@ -6824,7 +6824,16 @@ def argsort(a, axis=np._NoValue, kind=None, order=None, endwith=True, fill_value
...
+    and applying MaskedArray sort() method.
+
+    See Also
+    --------
+    numpy.ma.MaskedArray.sort()
```
Comment (anirudh2290) [inline numpy/ma/core.py:6835]:
```suggestion
    MaskedArray.sort : equivalent function
```
to make it consistent with other docstrings.

**Review by anirudh2290 — APPROVED** [review]
Just pointed out two nits, overall looks good!

File: numpy/ma/core.py (hunk 3 above)
Comment (vrakesh) [reply numpy/ma/core.py]: Addressed the changes.

File: numpy/ma/core.py [hunk 4]
```diff
@@ -6824,7 +6824,16 @@ def argsort(a, axis=np._NoValue, kind=None, order=None, endwith=True, fill_value
 argsort.__doc__ = MaskedArray.argsort.__doc__
 
 def sort(a, axis=-1, kind=None, order=None, endwith=True, fill_value=None):
-    "Function version of the eponymous method."
+    """
+    Return a sorted copy of the masked array.
+
+    Equivalent to creating a copy of the array
+    and applying MaskedArray sort() method.
```
Comment (seberg) [inline numpy/ma/core.py:6831]:
```suggestion
    and applying the MaskedArray ``sort()`` method.
```
I am wondering if "Refer to `MaskedArray.sort` for the full documentation" in some form may be better/necessary, mainly.

File: numpy/ma/core.py [hunk 5]
```diff
@@ -6824,7 +6824,16 @@ def argsort(a, axis=np._NoValue, kind=None, order=None, endwith=True, fill_value
...
+
+    See Also
+    --------
+    MaskedArray.sort : equivalent function
```
Comment (seberg) [inline numpy/ma/core.py:6835]:
```suggestion
    MaskedArray.sort : equivalent method
```
?

**Review by seberg — APPROVED** [review]
Anyway, we are starting to nitpick over small stuff, LGTM in general.

File: numpy/ma/core.py (hunk 5 above)
Comment (vrakesh) [reply numpy/ma/core.py]: Done

**Comment by seberg** [comment]
Thanks, lets put this in!

**Pull request merged**